### PR TITLE
reform c array code

### PR DESCRIPTION
**Pull request by danielhu19**
If this PR is related to coding or code translation, please read the [contribution guideline](https://github.com/krahets/hello-algo/issues/15#issue-1464069591), fill out the checklist, and paste the console outputs to the PR.

- [✅] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [✅] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [✅] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/c/include/macro_fun.h
Comment (danielhu19) [inline codes/c/include/macro_fun.h:1]: highly recommended

File: codes/c/chapter_array_and_linkedlist/array.c [hunk 1]
```diff
@@ -76,37 +93,41 @@ int main() {
     printf("数组 arr = ");
     printArray(arr, size);
 
-    int nums[5] = { 1, 3, 2, 5, 4 };
+    int nums[5] = { 1, 3, 3, 5, 4 };
```
Comment (krahets) [inline codes/c/chapter_array_and_linkedlist/array.c:96]: Please do not modify the test case

**Review by danielhu19 — COMMENTED** [review]
- 精修了这一小部分
- 结合更多C语言的特性

File: codes/c/chapter_array_and_linkedlist/array.c [hunk 2]
```diff
@@ -16,55 +16,72 @@ int randomAccess(int* nums, int size) {
...
+    nums = extend(nums, size , 1);
     // 把索引 index 以及之后的所有元素向后移动一位
-    for (int i = size - 1; i > index; i--) {
+    for (int i = *size - 1; i > index; i--) {
         nums[i] = nums[i - 1];
     }
     // 将 num 赋给 index 处元素
     nums[index] = num;
+
+    return nums;
```
Comment (krahets) [inline codes/c/chapter_array_and_linkedlist/array.c:50]: Why return `nums`? If it makes sense, why do not return `nums` in `removeItem()`?

File: codes/c/chapter_array_and_linkedlist/array.c [hunk 3]
```diff
@@ -76,37 +93,41 @@ int main() {
...
-    /* 遍历数组 */
-    traverse(res, size);
+    /* 遍历数组(直接通过下标) */
     
     /* 查找元素 */
-    int index = find(res, size, 3);
-    printf("在 res 中查找元素 3 ，得到索引 = %d\n", index);
+    int *index = find(res, size, 3, compare);
+    printf("在 res 中查找元素 3 ，得到索引 = ");
+    // ultimate answer 🤪
```
Comment (krahets) [inline codes/c/chapter_array_and_linkedlist/array.c:125]: Remove the unrelated codes

File: codes/c/chapter_array_and_linkedlist/array.c (hunk 1 above)
Comment (danielhu19) [reply codes/c/chapter_array_and_linkedlist/array.c:96]: > All the code languages are consistent in terms of the test cases, so please do not modify them.
I might need to close the pr cause I forgot to consider the compatibility with the content. Btw, thanks for replying!

**Pull request closed without merging**